result.go

```diff
@@ -93,18 +107,20 @@ func (a *Application) WhenResult(serviceID string, conditions ...ResultCondition
 // It's possible to add multiple filters by calling Filter multiple times.
 // Other filter funcs and the task execution will no proceed if a filter
 // func returns false.
-func (e *ResultEmitter) Filter(fn func(*Result) bool) *ResultEmitter {
-	e.m.Lock()
-	defer e.m.Unlock()
+func (e *ResultEmitter) Filter(fn func(*Result) (execute bool)) *ResultEmitter {
 	e.filterFuncs = append(e.filterFuncs, fn)
 	return e
 }
 
+// SetTags sets execution tags for task executions.
+func (e *ResultEmitter) SetTags(fn func(*Result) (tags []string)) *ResultEmitter {
```

Comment:
@ilgooz I'm not ok with this function here.
It will be weird to use it. Eg:
```
WhenXXX()
.Filter()
.SetTags()
.Map()
.Execute()
```

What about adding option to the execution functions?

Anyway, can you remove it from this PR so we can merge and we discuss and implement it in a new issue/PR